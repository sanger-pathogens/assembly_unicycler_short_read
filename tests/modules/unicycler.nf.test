nextflow_process {

    name "Test Process UNICYCLER"
    script "modules/unicycler.nf"
    process "UNICYCLER"

    test("Should run without failures") {

        when {
            params {
                max_cpus   = "1"
                max_memory = "2.GB"
                max_time   = "1.h"
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = [["id": "test_id"], [file("test_data/inputs/test_1.fastq.gz"), file("test_data/inputs/test_2.fastq.gz")]]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.scaffolds, process.out.gfa).match()
        }

    }

}
