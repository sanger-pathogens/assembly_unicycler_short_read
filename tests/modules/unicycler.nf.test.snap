{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test_id"
                    },
                    "test_id.scaffolds.fa:md5,845ae37a39ee6ea454f3830855cd450a"
                ]
            ],
            [
                [
                    {
                        "id": "test_id"
                    },
                    "test_id.assembly.gfa:md5,24044b979d80e9ca4da184c5d1775d6d"
                ]
            ]
        ],
        "timestamp": "2023-02-18T11:42:37+0000"
    }
}